### PR TITLE
fix OCA\DAV\CalDAV\CalDavBackend search $options

### PR DESCRIPTION
**Pull request by dartcafe**
Regarding https://github.com/nextcloud/polls/issues/411#issuecomment-570941255

### CODE REVIEW

**Comment by tcitworld** [comment]
Do you mind also adding a related test in `apps/dav/tests/unit/CalDAV/CalDavBackendTest.php`?

File: apps/dav/lib/CalDAV/CalDavBackend.php [hunk 1]
```diff
@@ -1553,12 +1553,12 @@ public function search(array $calendarInfo, $pattern, array $searchProperties,
 		if (isset($options['timerange'])) {
 			if (isset($options['timerange']['start'])) {
```
Comment (tcitworld) [inline apps/dav/lib/CalDAV/CalDavBackend.php]: We could also test if the value is a `DateTime`.

**Comment by dartcafe** [comment]
> 
> 
> Do you mind also adding a related test in `apps/dav/tests/unit/CalDAV/CalDavBackendTest.php`?

erks. I am not familiar with the tests.... That's a pending thing in polls. :-)

**Comment by dartcafe** [comment]
OK, I tried. Tell me, if this was OK.

**Comment by tcitworld** [comment]
Added two commits to address my comments.

**Comment by dartcafe** [comment]
As we support version 16-18, will there be a backport, so we can get rid of https://github.com/nextcloud/polls/pull/747?

**Comment by dartcafe** [comment]
@rullzer @georgehrke 
Any change to get this merged and backported, so we can release our new feature?

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Please rebase :)

**Review by tcitworld — APPROVED** [review]

**Comment by dartcafe** [comment]
any doa? Still waiting for seeing this in a release.

**Review by georgehrke — APPROVED** [review]

**Pull request merged**